api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/ConditionalStep.java

```diff
@@ -1,47 +1,69 @@
 package edu.cornell.mannlib.vitro.webapp.dynapi.components;
 
+import java.util.HashSet;
 import java.util.Set;
 
 import edu.cornell.mannlib.vitro.webapp.dynapi.OperationData;
+import edu.cornell.mannlib.vitro.webapp.dynapi.computation.StepInfo;
 
 public class ConditionalStep implements Step {
 
     private Condition condition;
+    
+    Step nextStepIfFailed = null;
+    Step nextStepIfSucceded = null;
 
-    @Override
-    public void dereference() {
-        // TODO Auto-generated method stub
-
-    }
 
     @Override
     public OperationResult run(OperationData input) {
         return null;
     }
 
     @Override
-    public Set<Link> getNextLinks() {
-        // TODO Auto-generated method stub
-        return null;
+    public Parameters getProvidedParams() {
+        return new Parameters();
+    }
+
+    public void setCondition(Condition condition) {
+        this.condition = condition;
     }
 
     @Override
-    public Parameters getRequiredParams() {
-        return condition.getRequiredParams();
+    public void dereference() {
+        // TODO Auto-generated method stub
     }
 
     @Override
-    public Parameters getProvidedParams() {
-        return new Parameters();
+    public Set<StepInfo> getNextNodes() {
+        HashSet<StepInfo> nextNodes = new HashSet<StepInfo>();
+        nextNodes.add(nextStepIfFailed);
```

Comment:
Will getNextNodes ever be called before setting nextStepIfFailed and nextStepIfSucceded? If so, wouldn't it be better to return an empty array rather than array of two null objects.

Reply:
Fixed by introduction NullStep